src/ODETools/AffineThetaMethod.jl

```diff
@@ -0,0 +1,93 @@
+function solve_step!(uf::AbstractVector,
+                     solver::ThetaMethod,
+                     op::AffineODEOperator,
+                     u0::AbstractVector,
+                     t0::Real,
+                     cache) # -> (uF,tF)
+
+  dt = solver.dt
+  solver.θ == 0.0 ? dtθ = dt : dtθ = dt*solver.θ
+  tθ = t0+dtθ
+
+  if cache === nothing
+    ode_cache = allocate_cache(op)
+    l_cache = nothing
+  else
+    ode_cache, l_cache = cache
+  end
+
+  ode_cache = update_cache!(ode_cache,op,tθ)
+
+  afop = ThetaMethodAffineOperator(op,tθ,dtθ,u0,ode_cache)
+
+  newmatrix = true
+  l_cache = solve!(uf,solver.ls,afop,l_cache,newmatrix)
+
+  if 0.0 < solver.θ < 1.0
+    uf = uf*(1.0/solver.θ)-u0*((1-solver.θ)/solver.θ)
+  end
+
+  cache = (ode_cache, l_cache)
+
+  tf = t0+dt
+  return (uf,tf,cache)
+
+end
+
+function solve_step!(uf::AbstractVector,
+                     solver::ThetaMethod,
+                     op::ConstantODEOperator,
+                     u0::AbstractVector,
+                     t0::Real,
+                     cache) # -> (uF,tF)
+
+  dt = solver.dt
+  solver.θ == 0.0 ? dtθ = dt : dtθ = dt*solver.θ
+  tθ = t0+dtθ
+
+  if cache === nothing
+    ode_cache = allocate_cache(op)
+    afop = ThetaMethodAffineOperator(op,tθ,dtθ,u0,ode_cache)
+    l_cache = nothing
+  else
+    ode_cache, vθ, afop, l_cache = cache
+  end
+
+  ode_cache = update_cache!(ode_cache,op,tθ)
+
+  residual!(afop.vector,odeop,tθ,u0,u0,ode_cache)
```

Comment:
I would keep a first version that it is true that involves a minor performance hit, and create an issue for thinking about the extension. I am not sure we have all we need, since we need to do the _linear assembly_ process twice, one of the mass and other for the stifness. I have to check `Gridap` for this